code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -576,3 +576,9 @@
 	if (AIStatus == AI_Z_OFF)
 		SSidlenpcpool.idle_mobs_by_zlevel[old_z] -= src
 		toggle_ai(initial(AIStatus))
+
+/mob/living/simple_animal/Destroy()
```

Comment:
Instead of defining a third `/mob/living/simple_animal/Destroy()`, could you combine them all into one `Destroy` proc?